Web Inspector: FIXME: Only pass RemoteObjects here so we can avoid this work

francescorn opened the pull request:
#### 767560c9dec712b630f91f535a193cc90c4a7065
<pre>
Web Inspector: FIXME: Only pass RemoteObjects here so we can avoid this work
<a href="https://bugs.webkit.org/show_bug.cgi?id=290972">https://bugs.webkit.org/show_bug.cgi?id=290972</a>

Reviewed by NOBODY (OOPS!).

Fix small grammar error in didSampleProbe.
Enforce that all callers pass only WI.RemoteObject instances by using console.assert.
Ensures _createRemoteObjectIfNeeded always receives correct type, allows the removal of unnecessary conversion code, and catches errors.

* LayoutTests/inspector/debugger/didSampleProbe-multiple-probes.html:
* Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js:
(WI.ConsoleMessageView.prototype._createRemoteObjectIfNeeded):
</pre>

<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ef295baf9548c18a6fe8a561ac55b1d829b730f7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/137366 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/9726 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/48653 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/145117 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/90339 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/10430 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/9853 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/105035 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/76733 "Exiting early after 60 failures. 21518 tests run. 60 failures") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/140311 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7710 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/123109 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85891 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/7347 "Passed tests") | [❌ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/5064 "Found 14 new API test failures: TestWebKitAPI.WKWebExtensionAPICookies.ErrorsRead, TestWebKitAPI.SnapshotStore.ClearSiteDataHeaderCrossOrigin, TestWebKitAPI.WKWebExtensionAPIStorage.SetAccessLevelTrustedContexts, TestWebKitAPI.SOAuthorizationPopUp.InterceptionSucceedNewWindowNavigation, TestWebKitAPI.Notification.RequestPermissionDenied, TestWebKitAPI.NSResponderTests.ValidRequestorForSendAndReturnTypes, TestWebKitAPI.WEBKIT.NavigationActionHasOpener12, TestWebKitAPI.WKWebExtensionAPIDOM.OpenOrClosedShadowRootViaElement, TestWebKitAPI.Notification.ParallelPermissionRequestsDenied, TestWebKitAPI.WKWebExtensionAPICookies.ErrorsWrite ... (failure)") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/5704 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/116712 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/41264 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/147874 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/9409 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41822 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/113409 "Found 1 new test failure: inspector/debugger/didSampleProbe-multiple-probes.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/9427 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/7919 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113750 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/7267 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/119352 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/64000 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/9458 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/37405 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/9188 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/73023 "Built successfully") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/9398 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/9250 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

francescorn committed:
Web Inspector: FIXME: Only pass RemoteObjects here so we can avoid this work
https://bugs.webkit.org/show_bug.cgi?id=290972

Reviewed by NOBODY (OOPS!).

Switch the order of this._message.target and parameter

* LayoutTests/inspector/debugger/didSampleProbe-multiple-probes.html:
* Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js:
(WI.ConsoleMessageView.prototype._createRemoteObjectIfNeeded):

francescorn committed:
Web Inspector: FIXME: Only pass RemoteObjects here so we can avoid this work
https://bugs.webkit.org/show_bug.cgi?id=290972

Reviewed by NOBODY (OOPS!).

Switch the order of this._message.target and parameter

* Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js:
(WI.ConsoleMessageView.prototype._createRemoteObjectIfNeeded):

francescorn committed:
Web Inspector: FIXME: Only pass RemoteObjects here so we can avoid this work
https://bugs.webkit.org/show_bug.cgi?id=290972

Reviewed by NOBODY (OOPS!).

Replace this._message.target, parameter with parameter, this._message.target

* Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js:
(WI.ConsoleMessageView.prototype._createRemoteObjectIfNeeded):

rcaliman-apple commented on Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js:
```diff
@@ -549,12 +549,11 @@ WI.ConsoleMessageView = class ConsoleMessageView extends WI.Object
 
         if (typeof parameter === "object")
-            return WI.RemoteObject.fromPayload(parameter, this._message.target);
+            return WI.RemoteObject.fromPayload(this._message.target, parameter);
```
This change is not necessary. The argument order of `WI.RemoteObject.fromPayload()` didn't change (and doesn't need to change).

rcaliman-apple commented on LayoutTests/inspector/debugger/didSampleProbe-multiple-probes.html:
```diff
@@ -37,9 +40,9 @@
+            message: "SampleIds from any probe are sequential and start counting from one.",
             predicate: function samplesHaveSequentialIds() {
-                let initialSampleId = samples[0].sampleId;
+                let initialSampleId = sampleIdCounter;  // Use the counter to ensure starting from 1
```
The changes in this test file seem unrelated. The test also fails after this change.

You can run tests locally to check:

```
run-webkit-tests --no-build --no-retry-failures --time-out-ms=5000 LayoutTests/inspector/debugger/didSampleProbe-multiple-probes.html
```

rcaliman-apple replied:
I was confused why the test failure didn't show up in EWS table at the top. Turns out this test is skipped on Mac configurations:

https://github.com/WebKit/WebKit/blob/39104faa73bc82f25ab31998a7bf842f8701600e/LayoutTests/platform/mac/TestExpectations#L495-L496 


https://trac.webkit.org/wiki/TestExpectations#Semantics
>If no expectation keyword is specified, then the Skip keyword is implied for compatibility with the (now obsolete) Skipped files.

Extra reasons not to make changes to it. (Unless you want to identify why it's skipped and fix the cause, but that's another patch)

The test is run on the Linux GTK configuration and fails with this change:
https://ews-build.webkit.org/#/builders/1/builds/83627

rcaliman-apple reviewed (changes requested):
What are you trying to achieve?

The FIXME comment is vague in this context. But the changes don't address it in any way.

francescorn committed:
Web Inspector: FIXME: Only pass RemoteObjects here so we can avoid this work
https://bugs.webkit.org/show_bug.cgi?id=290972

Reviewed by NOBODY (OOPS!).

null and undefined are now handled before the "object" check
Prevents null from being incorrectly passed to fromPayload

* LayoutTests/inspector/debugger/didSampleProbe-multiple-probes.html:
* Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js:
(WI.ConsoleMessageView.prototype._createRemoteObjectIfNeeded):

francescorn committed:
Web Inspector: FIXME: Only pass RemoteObjects here so we can avoid this work
https://bugs.webkit.org/show_bug.cgi?id=290972

Reviewed by NOBODY (OOPS!).

null and undefined are now handled before the "object" check
Prevents null from being incorrectly passed to fromPayload
(Fix small grammar error)

* LayoutTests/inspector/debugger/didSampleProbe-multiple-probes.html:
* Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js:
(WI.ConsoleMessageView.prototype._createRemoteObjectIfNeeded):

francescorn committed:
Web Inspector: FIXME: Only pass RemoteObjects here so we can avoid this work
https://bugs.webkit.org/show_bug.cgi?id=290972

Reviewed by NOBODY (OOPS!).

Fix small grammar error in didSampleProbe
null and undefined are now handled before the "object" check
Prevents null from being incorrectly passed to fromPayload

* LayoutTests/inspector/debugger/didSampleProbe-multiple-probes.html:
* Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js:
(WI.ConsoleMessageView.prototype._createRemoteObjectIfNeeded):

francescorn committed:
Web Inspector: FIXME: Only pass RemoteObjects here so we can avoid this work
https://bugs.webkit.org/show_bug.cgi?id=290972

Reviewed by NOBODY (OOPS!).

Fix small grammar error in didSampleProbe
Add a console.assert to pass only RemoteObjects in _createRemoteObjectIfNeeded and return the parameter

* LayoutTests/inspector/debugger/didSampleProbe-multiple-probes.html:
* Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js:
(WI.ConsoleMessageView.prototype._createRemoteObjectIfNeeded):
